### PR TITLE
realize my internet shop with null pointer exc

### PR DESCRIPTION
**Pull request by RostykRosolovskyi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,12 +1,92 @@
...
+            <groupId>javax.servlet</groupId>
+            <artifactId>jstl</artifactId>
+            <version>1.2</version>
+        </dependency>
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.17</version>
+        </dependency>
+
```
Comment (Sofasmile) [inline pom.xml]: remove two redundant lines

File: src/main/java/internetshop/controller/AddToBucketController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    private static UserService userService;
+    @Inject
+    private static BucketService bucketService;
+    @Inject
+    private static ItemService itemService;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        User user = userService.get(0L);
```
Comment (Sofasmile) [inline src/main/java/internetshop/controller/AddToBucketController.java]: 0L is the magic number?
make constant from it in all controllers

File: src/main/java/internetshop/controller/CompleteOrderController.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long bucketId = Long.parseLong(req.getParameter("bucket_id"));
+        Bucket bucket = bucketService.get(bucketId);
+        if (bucket.getItems().size() != 0) {
+            User user = userService.get(0L);
+            Order order = new Order(bucket.getItems(), user);
+            orderService.add(order);
+            user.getOrders().add(order);
```
Comment (Sofasmile) [inline src/main/java/internetshop/controller/CompleteOrderController.java:35]: we can move this line in method `completeOrder` in the service layer, and then call this method here

File: src/main/java/internetshop/controller/GetBuckectController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        User user = userService.get(0L);
+        Bucket bucket = bucketService.get(user.getBucket().getId());
+        req.setAttribute("user", user);
+        req.setAttribute("bucket", bucket);
+        List<Item> items = bucket.getItems();
+        double sum = items.stream()
```
Comment (Sofasmile) [inline src/main/java/internetshop/controller/GetBuckectController.java]: do we need it here?

File: src/main/java/internetshop/controller/ShowOrdersController.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    @Inject
+    private static OrderService orderService;
+    @Inject
+    private static UserService userService;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        User user = userService.get(0L);
+        List<Order> orders = user.getOrders();
```
Comment (Sofasmile) [inline src/main/java/internetshop/controller/ShowOrdersController.java]: we can use method getOrders()

File: src/main/webapp/WEB-INF/views/bucket.jsp [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    <tr>
+        <td>Total</td>
+        <td>${totalCost}</td>
+        <td><a href="completeOrder?bucket_id=${bucket.id}">Buy</a></td>
+    </tr>
+</table>
+<br>
+<a href="shop">Return to the shop</a>
+</body>
+</html>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/bucket.jsp]: add a blank line

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    <hr>
+    <label for="name"><b>First name</b></label>
+    <input type="text" placeholder="first name" name="name" id="name" required>
+    <label for="surname"><b>Last name</b></label>
+    <input type="text" placeholder="last name" name="surname" id="surname" required>
+    <hr>
+    <button type="submit">Enter</button>
+</form>
+</body>
+</html>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/index.jsp]: the same

File: src/main/webapp/WEB-INF/views/orders.jsp [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+</table>
+
+<br>
+
+<a href="bucket">Show user's bucket</a>
+<br>
+<a href="shop">Return to the shop</a>
+
+</body>
+</html>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/orders.jsp]: :)

File: src/main/webapp/WEB-INF/views/register.jsp [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+            <hr>
+
+            <button type="submit">Register</button>
+        </div>
+        <a href="users">Users list</a><br>
+        <a href="index">To the main page</a>
+
+    </form>
+</body>
+</html>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/register.jsp]: :)

File: src/main/webapp/WEB-INF/views/shop.jsp [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+            <td><a href="addToBucket?item_id=${item.id}">Add</a></td>
+        </tr>
+    </c:forEach>
+</table>
+<br>
+<a href="bucket">Show user's bucket</a>
+<br>
+<a href="showAllOrders">Show user's orders</a>
+</body>
+</html>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/shop.jsp]: :)

File: src/main/java/internetshop/controller/InjectInitializer.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+public class InjectInitializer implements ServletContextListener {
+
+    private static final Logger logger = Logger.getLogger(InjectInitializer.class);
+
+    @Override
+    public void contextInitialized(ServletContextEvent sce) {
+        try {
+            Injector.injectDependencies();
+        } catch (IllegalAccessException e) {
+            logger.error(e);
```
Comment (Sofasmile) [inline src/main/java/internetshop/controller/InjectInitializer.java]: try to add some description

File: src/main/java/internetshop/lib/Injector.java [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+public class Injector {
+    private static List<Class> classes = new ArrayList<>();
+    public static final String PROJECT_MAIN_PACKAGE = "internetshop";
+
+    static {
+        try {
+            classes.addAll(getClasses(PROJECT_MAIN_PACKAGE));
+        } catch (ClassNotFoundException | IOException e) {
+            e.printStackTrace();
```
Comment (Sofasmile) [inline src/main/java/internetshop/lib/Injector.java]: use logger here

File: src/main/java/internetshop/controller/CompleteOrderController.java (hunk 3 above)
Comment (RostykRosolovskyi) [reply src/main/java/internetshop/controller/CompleteOrderController.java:35]: i will rewrite this in next homework ,when i will use session, ok?

File: src/main/java/internetshop/controller/AddToBucketController.java [hunk 13]
```diff
@@ -21,11 +22,12 @@
     private static BucketService bucketService;
     @Inject
     private static ItemService itemService;
+    private static final Long sesionId = 0L;
```
Comment (Sofasmile) [inline src/main/java/internetshop/controller/AddToBucketController.java:25]: https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names
```suggestion
    private static final Long USER_SESSION_ID = 0L;
```

File: src/main/java/internetshop/controller/CompleteOrderController.java (hunk 3 above)
Comment (Sofasmile) [reply src/main/java/internetshop/controller/CompleteOrderController.java:35]: I will check it in nest hw :)

**Review by Sofasmile — APPROVED** [review]
make small fixes
I will check them in next hw)

**Pull request merged**